### DISCUSSION TITLE
text/html turns into application/octet-stream after saving edits

### DISCUSSION THREAD

**Discussion by heyycap** [iterate-ch/cyberduck#15734] [Q&A]
Cyberduck Version 8.8.1 (41263)
Editor: Sublime Text 4, Build 4169
Protocol: AWS
OS: macOS Ventura 13.6.3

I have a website through AWS (Route 53, S3, Cloudfront). Worth noting is that I like to omit ".html" from my html files for a cleaner url. In the metadata, `Content-Type` is set to `text/html` so it's served instead of just downloaded. I also have the `<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">` tag in the file's `<head>`.

### Steps

1. Edit html file with Sublime Text 4 (cmd+k)
2. Save edits (Cyberduck notifies me the file is uploaded)
3. Create invalidation in Cloudfront so cached version of that file is removed
4. Upon visiting edited page, page is downloaded instead of served
5. In Amazon S3 bucket, metadata for edited file has changed to `Content-Type: application/octet-stream`

Optionally, immediately after step 2, I can skip to step 5 to see the change in metadata.

### Expected Behavior

On step 4, I expect the page to be served like before. On step 5, I expect the metadata to stay as `text/html` instead of changing.

### What I've Tried

I don't experience the same issue after editing my index.html file.

I also notice that I can't view or edit the `Content-Type` metadata attribute in Cyberduck (cmd+i).

After testing with previous versions of Cyberduck, I found the latest version I can edit suffixless html files (without the metadata changing on me) and the `Content-Type` property is 8.7.2. With >= 8.8.0, I run into the above issues. I'm going to stick with 8.7.2 for now, but it would be great to know if this is how 8.8.0+ is supposed to work, or if there's something I need to do to make the metadata stick.

**Comment by dkocher** [comment]
Possibly a regression from 51d46a06.

**Comment by dkocher** [comment]
We are now tracking this issue in #15742.
